packages/unlock-js/src/uniswapService.ts

```diff
@@ -0,0 +1,150 @@
+import { NetworkConfigs } from '@unlock-protocol/types'
+import { ethers } from 'ethers'
+import { GraphQLClient } from 'graphql-request'
+import { abi as UniswapV3FactoryABI } from '@uniswap/v3-core/artifacts/contracts/UniswapV3Factory.sol/UniswapV3Factory.json'
+
+export interface PoolOptions {
+  network: number
+  tokenIn: string
+  tokenOut: string
+  fee?: number
+}
+
+interface PriceOptions {
+  network?: number
+  baseToken: string
+  quoteToken: string
+  amount: number
+}
+
+export class UniswapService {
+  constructor(public networks: NetworkConfigs = networks) {}
+
+  getNetworkConfigAndProvider(networkId: number) {
+    const networkConfig = this.networks[networkId]
+    if (!networkConfig) {
+      throw new Error(`No network found for the id: ${networkId}`)
+    }
+    const provider = new ethers.providers.JsonRpcProvider(
+      networkConfig.provider
+    )
+    return {
+      networkConfig,
+      provider,
+    }
+  }
+
+  /**
+   * Get pool address for erc20 token in and token out pairs on uniswap. By default, we find the 5% fee pools.
+   */
+  async getPoolAddress({
```

Comment:
Looking at the subgraph I think we can probably get the pool from there?

Reply:
As mentioned on the discord. It does not work for some tokens it should hence the replacement with on-chain calls. 

The where filter is unreliable.